prism_compile.c

```diff
@@ -787,6 +787,44 @@ pm_interpolated_node_compile(pm_node_list_t *parts, rb_iseq_t *iseq, NODE dummy_
     if (parts_size > 0) {
         VALUE current_string = Qnil;
 
+        bool literal = true;
+        for (size_t index = 0; index < parts_size; index++) {
+            const pm_node_t *part = parts->nodes[index];
+
+            if (!PM_NODE_TYPE_P(part, PM_STRING_NODE)) {
+                literal = false;
+                break;
+            }
+        }
+
+        if (literal) {
+            for (size_t index = 0; index < parts_size; index++) {
```

Comment:
I'm not sure I understand why this string part gluing-together logic is duplicated in the literal and the section below?

Reply:
This is change is not strictly related to this PR, it's just to fix a prism issue that happens now that we feed `--enable-frozen-literal / disable-frozen-literal` to prism.

Prism parses `a = "foo""bar"` as string interpolation nodes, even though it's just a static string. Interpolated strings are always mutable, never frozen. So we need to do all this to figure out if we need to produce a frozen/mutabe/chilled string.

I could refactor to share this code, but I think @kddnewton wants to change how it's parsed, so not too sure if it's worth it.

Reply:
One thought about this is maybe `PM_NODE_FLAG_STATIC_LITERAL` could be used instead of the loop above?
Wouldn't simplify that much though.

Reply:
Ah, didn't notice this. But yeah, probably not worth bothering.